src/main/proto/control.proto

```diff
@@ -223,4 +241,33 @@ message EventProcessorSegmentReference {
 }
 
 message Heartbeat {
+}
+
+message TopologyChange {
+    UpdateType update_type  = 1;
+    string context = 2;
+    string client_id = 3; // The client ID of the component requesting the update
+    string client_stream_id = 4; // The client ID of the component requesting the update
+    string component_name = 5; // The name of the component requesting the update
+    oneof subscription {
+        CommandSubscription command = 6; // The subscription ID to add or remove
+        QuerySubscription query = 7; // The processor name to add or remove
+    }
+}
+
+message CommandSubscription {
+    string name = 1; // The subscription ID to add or remove
+    int32 load_factor = 2; // The load factor for the subscription
+}
+
+message QuerySubscription {
+    string name = 1; // The subscription ID to add or remove
+}
+
+enum UpdateType {
+    ADD_COMMAND_HANDLER = 0; // a new command handler has been added
+    REMOVE_COMMAND_HANDLER = 1; // a command handler has been removed
+    ADD_QUERY_HANDLER = 2; // a new query handler has been added
+    REMOVE_QUERY_HANDLER = 3; // a query handler has been removed
+    RESET_ALL = 4; // reset all information in the client, new information will be sent
```

Comment:
Curious: When would a "reset all" occur?

Reply:
RESET_ALL is sent to a listener when the client loses the connection to Axon Server.